converter/diag/diagnostics.go

```diff
@@ -81,3 +82,24 @@ func (ds Diagnostics) Error() string {
 
 	return sb.String()
 }
+
+func (ds Diagnostics) GenerateReport(reportType string, writer io.Writer) error {
+	switch reportType {
+	case Text:
+		return generateTextReport(ds, writer)
+	default:
+		return fmt.Errorf("unsupported diagnostic report type %q", reportType)
+	}
+}
```

Comment:
I may be making this up, but I think usually io.Writer is the first argument to a function like this (a la Fprint)